pydicom/filereader.py

```diff
@@ -115,15 +116,10 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
-    tag_set = set()
-    if specific_tags is not None:
-        for tag in specific_tags:
-            if isinstance(tag, str):
-                tag = Tag(tag_for_keyword(tag))
-            if isinstance(tag, BaseTag):
-                tag_set.add(tag)
-        tag_set.add(Tag(0x08, 0x05))
-    has_tag_set = len(tag_set) > 0
+    tag_set = {Tag(tag) for tag in specific_tags} if specific_tags else []
+    has_tag_set = bool(tag_set)
```

Comment:
`else {}` ?  I didn't look through to see whether it matters (probs not used if empty), but seems better to keep same type.

Reply:
Changed to `set()`, You're right, it'd be better to keep the type consistent in case of later modifications.